calculator_test.go

```diff
@@ -7,162 +7,176 @@ import (
 	"testing"
 )
 
-type testCase struct {
-	a, b        float64
-	want        float64
-	comment     string
-	err         error
-	errExpected bool
-}
-
-var testCasesAdd = []testCase {
-	{a: 1, b: 2, want: 3, comment: "one add two"},
-	{a: 2, b: 5, want: 7, comment: "two add five"},
-}
-
-var testCasesSubtract = []testCase{
-	{a: 1, b: 2, want: -1, comment: "one minus two"},
-	{a: 2, b: 5, want: -3, comment: "two minus five"},
-}
-
-var testCasesMultiply = []testCase{
-	{a: 1, b: 2, want: 2, comment: "one times two"},
-	{a: 2, b: 5, want: 10, comment: "two times five"},
-}
-
-var testCasesDivide = []testCase{
-	{a: 2, b: 1, want: 2, errExpected: false, comment: "two divided by 1"},
-	{a: 10, b: 5, want: 2, errExpected: false, comment: "ten divided by 5"},
-	{a: 10, b: 0, want: 0, errExpected: true, comment: "error out on divide by zero"},
-}
-
-var testCasesSqrt = []testCase{
-	{a: 16, want: 4, errExpected: false, comment: "square root of 16"},
-}
-
 func TestAdd(t *testing.T) {
+	testCases := []struct {
+		a, b    float64
+		want    float64
+		comment string
+	}{
+		{a: 1, b: 2, want: 3, comment: "one add two"},
+		{a: 2, b: 5, want: 7, comment: "two add five"},
```

Comment:
Let's add a few more challenging test cases! We're learning how to think adversarially about our code: what might possibly break it? Negative plus positive, with a negative result? With a positive result? A zero result? Negative plus negative? Infinity? Fractional values?

Reply:
sorted, thanks man